igramarea.cpp

```diff
@@ -2095,6 +2102,100 @@ void IgramArea::CenterOutlineActive(bool checked){
+
+    // note that I (gr5) tested what happens if you have non-json in the oln file.  For example missing quotes commas and brackets.
+    // basically you get zeroes.  Zeroes in outlines, zero polygons, etc.  This causes no harm as it's basically like not having
+    // an oln file at all (or some portions of the oln data get lost).  Because of this I decided not to report any json parsing errors.
+    QJsonDocument loadDoc(QJsonDocument::fromJson(saveData));
+
+    // outer
+    QJsonObject outside = loadDoc["outside_outline"].toObject();
+    CircleOutline out(outside);
+    m_outside = out;
+    m_outside.translate(QPointF(-cropTotalDx, -cropTotalDy));
+    m_OutterP1 = m_outside.m_p1.m_p;
+    m_OutterP2 = m_outside.m_p2.m_p;
+    outterPcount = 2;
+
+    // center/inner
+    QJsonObject inside = loadDoc["inside_outline"].toObject();
+    CircleOutline in(inside);
+    in.translate(QPointF(-cropTotalDx, -cropTotalDy));
+    m_center = in;
+    if (m_center.m_radius > 0) {
+        m_innerP1 = m_center.m_p1.m_p;
+        m_innerP2 = m_center.m_p2.m_p;
+        innerPcount = 2;
+    }
+
+    // filter
+    double filt = loadDoc["dft_filter_radius"].toDouble();
+    emit dftCenterFilter(filt);
+
+
+    // edge mask
+    QJsonValue jedge = loadDoc["edge_mask_width"];
+    mirrorDlg &md = *mirrorDlg::get_Instance();
+    if (jedge.isDouble()) {
+        double edge = jedge.toDouble();
+        // if outline edge mask is different than current ask user
+        if (edge != md.aperatureReduction){
+            QString text(
+                        "Do you want to change the mirror config value to match the value in the outline file?\n"
+                        "If no then the current mirror config value will be used instead."
+                         );
+
+            QMessageBox mb;
+            mb.setText(QString("Edge mask value in outline file for this interferogram is %1 and is different than mirror config value of %2.").arg(
+                                         edge, 6, 'f', 1).arg(md.aperatureReduction, 6, 'f', 1) );
+            mb.setInformativeText(text);
+            mb.setStandardButtons( QMessageBox::Yes|QMessageBox::No );
+            mb.setWindowTitle("Existing Interferogram outline file and Mirror Config difference.");
+            int width = QGuiApplication::primaryScreen()->geometry().width() * .5;
+            QSpacerItem* horizontalSpacer = new QSpacerItem(width, 0, QSizePolicy::Minimum, QSizePolicy::Expanding);
+            QGridLayout* layout = (QGridLayout*)mb.layout();
+            layout->addItem(horizontalSpacer, layout->rowCount(), 0, 1, layout->columnCount());
+            int resp = mb.exec();
+
+            switch (resp){
+            case QMessageBox::Yes:
+                md.changeEdgeMaskvalues(edge);
+
+                break;
+            case QMessageBox::No:
+                md.changeEdgeMaskvalues(md.aperatureReduction);
+                break;
+            }
```

Comment:
```suggestion
            default:
                // TODO log unexpected case
                break;
            }
```
For readability and safety
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es79-use-default-to-handle-common-cases-only

Reply:
This is copy pasted from Dale's code where he reads in the older style outline file.  He wrote this code just a week or two ago I think.  I suppose there is a third possibility maybe where they hit the "X" button in the corner (if there is one - there might be one).  I'll go test it right now...

Reply:
There is indeed a red X and clicking it doesn't crash or anything that bad but I'll make it do one of the 2 behaviors.

Reply:
I made default do the same as "no" button and added a comment.